scripts/setup-minikube.sh

```diff
@@ -0,0 +1,354 @@
+
+  if [ "${RESULT}" -eq 0 ]; then
+    echo namespace kubeflow already exists
+  else
+    kubectl create namespace kubeflow
+  fi
+
+  #download_kfctl_scripts
+  KUBEFLOW_REPO=/usr/local/google/home/abhishek/kubeflow ./kfctl.sh init localapp --platform minikube
+  pushd .
+  cd localapp
+  ../kfctl.sh generate all
+  ../kfctl.sh apply all
+  popd
+
+  if is_kubeflow_ready
+  then
+    mount_local_fs
+    setup_tunnels
+  else
+    echo -e "${RED}Unable to get kubeflow ready${NC}"
+  fi
+}
+
+function is_kubeflow_ready() {
+#  echo -ne "${YELLOW}Downloading bootstrapper image."
+#  until [[ `kubectl -n kubeflow-admin get pod kubeflow-bootstrapper-0 | tail -1 | awk '{print $3}'` == "Running" ]]
+#  do
+#    echo -n "."
+#    sleep 30
+#  done
+#  echo -e "${GREEN}[OK]${NC}"
+
+  echo -en "${YELLOW}Getting kubeflow namespace ready...${NC}"
+  local ns_ready=false
+  for i in {1..5}
+  do
+    kube_ns=`kubectl get namespaces | grep kubeflow | wc -l`
+    # if [ $kube_ns = 2 ]
+    if [ $kube_ns = 1 ]  # for kfctl
+    then
+       ns_ready=true
+       echo -e "${GREEN}[OK]${NC}"
+       break
+    fi
+    sleep 30
+  done
+
+  if [[ $ns_ready = false ]]
+  then
+    return 1
+  fi
+
+  svc_ready=false
+  amb_up=0
+  tf_hub_up=0
+  echo -en "${YELLOW}Bringing kubeflow services up."
+  until (( "$amb_up" > 0 && "$tf_hub_up" > 0 ))
+  do
+    sleep 30
+    amb_up=`kubectl -n kubeflow get pods | grep Running | grep ambassador | wc -l`
+    tf_hub_up=`kubectl -n kubeflow get pods | grep Running | grep tf-hub | wc -l`
+    echo -n "."
+    if (( "$amb_up" > 0 && "$tf_hub_up" > 0 ))
+    then
+      svc_ready=true
+      echo -e "${GREEN}[OK]${NC}"
+      break
+    fi
+  done
+
+  if [ $svc_ready = false ]
+  then
+    return 1
+  fi
+}
+
+
+# if user requested a local fs path to be mounted, make it accessible via
+# Jupyter Notebooks
+function mount_local_fs() {
+  if $MOUNT_LOCAL
+  then
+    # Create a persistent volume
+    cat <<EOF > ./pv.yaml
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local-volume
+  labels:
+    type: local
+spec:
+  persistentVolumeReclaimPolicy: Delete
+  storageClassName: local-storage
+  capacity:
+    storage: 10Gi
+  accessModes:
+    - ReadWriteOnce
+  local:
+    path: "/mnt/local"
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - minikube
+EOF
+  kubectl create -f ./pv.yaml
+
+  # Create a PVC attached to the volume
+  cat <<EOF > ./pv-claim.yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: local-notebooks
+spec:
+  storageClassName: local-storage
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  volumeName: local-volume
+EOF
+  kubectl create -n kubeflow -f ./pv-claim.yaml
+
+  # update tf-hub stateful set env to use the claim
+  kubectl -n kubeflow set env statefulset tf-hub -e KF_PVC_LIST=local-notebooks
```

Comment:
Can we do this in kfctl.sh during the generate phase?
Can we check if platform is minikube and then set ksonnet parameters to set these parameters on tf-hub?